core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/mercury/v03/request.go

```diff
@@ -74,11 +74,10 @@ func (c *client) DoRequest(ctx context.Context, streamsLookup *mercury.StreamsLo
 		c.multiFeedsRequest(ctx, ch, streamsLookup)
 	})
 
-	// TODO (AUTO 9090): Understand and fix the use of context.Background() here
-	reqTimeoutCtx, cancel := context.WithTimeout(context.Background(), mercury.RequestTimeout)
```

Comment:
We had actually seen strange issues when we were creating a new context from existing context with timeout instead of context.background where we saw the timeout was not being respected. https://smartcontract-it.atlassian.net/browse/AUTO-9090 has more details

if we want to fix this ideally we should also test the behaviour to ensure it is working correctly. I am not sure what's the impact of letting this code be as is, is there a possibility to have an exception for use of background context here and not change this?

Reply:
I don't understand the issue described in the ticket. If the claim is that something continues to run _longer_ than the given timeout, that is not necessarily a bug, and leaving out the parent context here will have no effect.

Reply:
Also, the timeout is not being applied to the actual request, only below when waiting for the response. That seems incorrect. Should this be moved up before the threadCtrl.GoCtx call?